Source/Pawnmorphs/Esoteria/HPatches/Optional/PawnScaling.cs

```diff
@@ -115,14 +115,15 @@ private static bool TryGetPawnFrameSetPrefix(Pawn pawn)
         }
 
 
-        // Calculate the rendered size based on body size
+        // Calculate the scale multiplier based on Pawnmorpher's BodySize multiplier
+        // TODO: Add a toggle to allow scaling by any body size difference compared to normal instead
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static float GetScale(Pawn pawn)
         {
             if (_currentPawn != pawn)
             {
                 _currentPawn = pawn;
-                _currentScaledBodySize = Mathf.Sqrt(pawn.BodySize / pawn.RaceProps.baseBodySize);
+                _currentScaledBodySize = Mathf.Sqrt(StatsUtility.GetStat(pawn, PMStatDefOf.PM_BodySize, 300) ?? 1f);
```

Comment:
You need to offset it by the pawn's base body size, since the pawn may be 2.0 body size per default, so -20% body size would be 1.6 rather than 0.8.

For regular pawns, their base body size is 1.0, but for aliens it could be anything.

Reply:
Is PM_BodySize not a multiplier?  I was under the assumption that it was always 100% unless a pawn has size-changing mutations.

(incidentally, it would actually have to be `pawn.RaceProps.baseBodySize * pawn.ageTracker.CurLifeStage.bodySizeFactor`.  The missing life stage factor was why it was scaling babies and kids wrong)

Reply:
You are right.
I was recalling '_currentScaledBodySize' as being the final size, but that is still multiplied on to the current size, so it expects a scale.